src/gc.c

```diff
@@ -585,12 +585,20 @@ static void gc_sweep_foreign_objs(void)
 // GC knobs and self-measurement variables
 static int64_t last_gc_total_bytes = 0;
 
+// max_total_memory is a suggestion.  We try very hard to stay
+// under this limit, but we will go above it rather than halting.
 #ifdef _P64
 #define default_collect_interval (5600*1024*sizeof(void*))
 static size_t max_collect_interval = 1250000000UL;
+// Eventually we can expose this to the user/ci.
+static uint64_t max_total_memory = (uint64_t) 2 * 1024 * 1024 * 1024 * 1024 * 1024;
```

Comment:
2 exabytes? That seems a little excessive...

Reply:
The idea is that the 64 bit heuristics will be limited by the available memory and the 32 bit heuristics are limited by the available address space.  The 2 exabytes was arbitrary but hopefully large enough to not be limiting.